lib/ProductOpener/Display.pm

```diff
@@ -531,24 +531,34 @@ $lc : language code
 
 It also initializes a request object that is returned.
 
+=head3 Parameters
+
+=head4 (optional) Request object reference $request_ref
+
+This function may be passed an existing request object reference
+(e.g. pre-containing some fields of the request, like a JSON body).
+
+If not passed, a new request object will be created.
+
+
 =head3 Return value
 
 Reference to request object.
 
 =cut
 
-sub init_request() {
+sub init_request ($request_ref = {}) {
```

Comment:
In python if you use a dict as a default parameter, it will be shared between every call.
(so you prefer to initialize with "undefined" and then if request_ref is undefined, you create a new dict).

But this seems to work in perl (I've done a test !)